src/core/server/elasticsearch/client/mocks.ts

```diff
@@ -31,42 +31,56 @@ const createInternalClientMock = (): DeeplyMockedKeys<Client> => {
     '_events',
     '_eventsCount',
     '_maxListeners',
+    'constructor',
     'name',
     'serializer',
     'connectionPool',
     'transport',
     'helpers',
   ];
 
+  const getAllPropertyDescriptors = (obj: Record<string, any>) => {
+    const descriptors = Object.entries(Object.getOwnPropertyDescriptors(obj));
+    let prototype = Object.getPrototypeOf(obj);
+    while (prototype != null && prototype !== Object.prototype) {
+      descriptors.push(...Object.entries(Object.getOwnPropertyDescriptors(prototype)));
+      prototype = Object.getPrototypeOf(prototype);
+    }
+    return descriptors;
+  };
+
   const mockify = (obj: Record<string, any>, omitted: string[] = []) => {
-    Object.keys(obj)
-      .filter((key) => !omitted.includes(key))
-      .forEach((key) => {
-        const propType = typeof obj[key];
-        if (propType === 'function') {
+    const descriptors = getAllPropertyDescriptors(obj);
+    descriptors
+      .filter((descriptor) => !omitted.includes(descriptor[0]))
+      .forEach((descriptor) => {
+        const key = descriptor[0];
+        if (typeof descriptor[1].value === 'function') {
           obj[key] = jest.fn(() => createSuccessTransportRequestPromise({}));
-        } else if (propType === 'object' && obj[key] != null) {
-          mockify(obj[key]);
+        } else if (typeof obj[key] === 'object' && obj[key] != null) {
+          mockify(obj[key], omitted);
         }
       });
   };
 
   mockify(client, omittedProps);
 
-  // client got some read-only (getter) properties
-  // so we need to extend it to override the getter-only props.
-  const mock: any = { ...client };
```

Comment:
This doesn't work anymore with the changes to `@elastic/elasticsearch::Client`. For example, it was excluding `nodes` from being defined on the mock because it's a getter, now... Example of how getters work with restructuring: https://stackblitz.com/edit/getter-destructuring?file=index.ts